src/lib/ui/thermostat/ThermostatCard.tsx

```diff
@@ -0,0 +1,149 @@
+import classNames from 'classnames'
+import { useState } from 'react'
+import {
+  isThermostatDevice,
+  type ThermostatDevice,
+  type ThermostatDeviceProperties,
+} from 'seamapi'
+
+import { FanIcon } from 'lib/icons/Fan.js'
+import { OffIcon } from 'lib/icons/Off.js'
+import { DeviceImage } from 'lib/ui/device/DeviceImage.js'
+import { ClimateSettingStatus } from 'lib/ui/thermostat/ClimateSettingStatus.js'
+
+interface ThermostatCardProps {
+  device: ThermostatDevice
+}
+
+export function ThermostatCard({ device }: ThermostatCardProps): JSX.Element {
+  return (
+    <div className='seam-thermostat-card'>
+      <Content device={device} />
+    </div>
+  )
+}
+
+function Content(props: { device: ThermostatDevice }): JSX.Element | null {
+  const { device } = props
+
+  const [temperatureUnit, setTemperatureUnit] = useState<
+    'fahrenheit' | 'celsius'
+  >('fahrenheit')
+
+  const toggleTemperatureScale = (): void => {
+    setTemperatureUnit(
+      temperatureUnit === 'fahrenheit' ? 'celsius' : 'fahrenheit'
+    )
+  }
+
+  if (device == null || (device != null && !isThermostatDevice(device))) {
+    return null
+  }
+
+  const {
+    temperature_fahrenheit: temperatureFahrenheit,
+    temperature_celsius: temperatureCelsius,
+    current_climate_setting: currentClimateSetting,
+    is_fan_running: isFanRunning,
+    relative_humidity: relativeHumidity,
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+  } = device.properties as ThermostatDeviceProperties
+
+  return (
+    <div className='seam-thermostat-card-content'>
+      <div className='seam-thermostat-card-image-wrap'>
+        <DeviceImage device={device} />
+      </div>
+      <div className='seam-thermostat-card-details'>
+        <div className='seam-thermostat-heading-wrap'>
+          <h4 className='seam-thermostat-card-heading'>
+            {device.properties.name}
+          </h4>
+          <button
+            onClick={toggleTemperatureScale}
+            className='seam-thermostat-temperature-toggle'
+          >
+            <span className='seam-thermostat-temperature-toggle-label'>
+              {temperatureUnit === 'fahrenheit' ? t.fahrenheit : t.celsius}
+            </span>
+          </button>
+        </div>
+
+        <div className='seam-thermostat-properties'>
+          <div className='seam-thermostat-property-block'>
+            <p className='seam-thermostat-property-label'>{t.temperature}:</p>
+          </div>
+          <div className='seam-thermostat-property-block'>
+            <p className='seam-thermostat-property-value'>
+              {Math.trunc(
```

Comment:
Can we re-use this component? Maybe it needs a more general name? https://react.seam.co/?path=/docs/library-climatesettingstatus--docs

Reply:
Are you just thinking of something like `Temperature` with F, C, and unit props?

Reply:
Sure, Temperature sounds nice!